asserts/database.go

```diff
@@ -129,14 +129,18 @@ type RODatabase interface {
 // assertions in the database.
 type Checker func(assert Assertion, signingKey *AccountKey, roDB RODatabase, checkTime time.Time) error
 
+// A NoAuthorityChecker defines a check on a no-authority assertion.
+type NoAuthorityChecker func(assert Assertion, roDB RODatabase) error
```

Comment:
I agree that we should have checkTime in here for future expansion, and I've done that.

I'm not convinced that we can reasonably check `account-key-request` assertions specifically against that, though.  The `since` and `until` headers really specify the desired values of those headers in the resulting `account-key` assertion, and for example in the case of revocation you'll be sending an AKR whose `until` header is in the past even when it's first received.  Furthermore, we're intentionally storing AKR solely for future reference, so I think the considerations for checking the timestamps of this assertion are somewhat different from those for AK.  I therefore haven't implemented a check like this for AKR at this time.